### PR TITLE
Add an additional test to key fields

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

There are two main changes in this PR, (sorry they're combined - I can split if people prefer):
1. I noticed that `TopologyGenerator` was creating new topology files for all previous versions, rather than just the current version. (I think it may have been me that broke it anyway).
 - fixed topology generation
 - deleted duplicate expected topology files
 - filtered out `ksql.streams.state.dir` as it changes per test, making it hard to diff.
 - rebuilt `5_3_0_pre` expected topologies to drop `ksql.streams.state.dir`

Reviewers notes:
 Look at changes to:
 - `EndToEndEngineTestUtil.java`
 - `QueryTranslationTest.java`
 - `TopologyFileGenerator.java`
 - The diffs for the existing topology files are all either a full delete (if duplicate), or just removing the state store dir.

2. Added new tests to `key-fields.json`.
 - added test cases to check the topology of downstream queries (as these are affected by the keyfield stored in the metastore).
 - added test to cover another edge case.
 - I've also documented implications of current key field management in `key-field.json`, as much for my benefit as for others.

Reviewers notes:
 Look at changes to:
 - `key-field.json`

### Testing done 
None functional change

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json [hunk 1]
```diff
@@ -200,14 +293,93 @@
...
+    },
+    {
+      "name": "stream | initially set | no key change | key in value | aliasing + duplicate | legacy downstream query | no key change",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT foo as aliased, bar as foo FROM INPUT;",
+        "CREATE STREAM DOWNSTREAM AS SELECT aliased, foo FROM OUTPUT PARTITION BY aliased;"
+      ],
+      "inputs": [
+        {"topic": "OUTPUT", "key": "1" ,"value": {"ALIASED":1, "FOO": 2}}
```
Comment (agavra) [inline ksql-engine/src/test/resources/query-validation-tests/key-field.json]: Same question as the ones below, why not generate the data into `INPUT`?

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json [hunk 2]
```diff
@@ -112,15 +148,34 @@
...
+          "key field of output has INCORRECT type: should be 'INT' - (see Issue #3 above)"
         ],
         "sources": [
           {"name": "INPUT", "type": "stream", "keyField": null},
           {"name": "OUTPUT", "type": "table", "keyField": {"name": "KSQL_INTERNAL_COL_0", "schema": {"type": "STRING"}}}
         ]
       }
     },
+    {
+      "name": "stream | initially null | group by (-) | key in value | no aliasing | legacy downstream query",
```
Comment (agavra) [inline ksql-engine/src/test/resources/query-validation-tests/key-field.json]: I'm not sure I understand what is "legacy" about downstream?

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json [hunk 3]
```diff
@@ -481,13 +729,34 @@
...
         "sources": [
           {"name": "OUTPUT", "type": "table", "keyField": {"name": "KSQL_INTERNAL_COL_0", "schema": {"type": "STRING"}}}
         ]
       }
     },
+    {
+      "name": "table | initially null | group by (-) | key in value | no aliasing | legacy downstream query",
+      "statements": [
+        "CREATE TABLE INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE TABLE INTERMEDIATE AS SELECT * FROM INPUT;",
```
Comment (agavra) [inline ksql-engine/src/test/resources/query-validation-tests/key-field.json]: what value does `INTERMEDIATE` add above the tests outlined above?

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json (hunk 2 above)
Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/key-field.json]: When I add a fix for these issues these 'legacy downstream' queries should continue to produce the same, incorrect, state and messages. Thereby proving/testing I've not broken/changed persistent queries running on customers clusters that were started in previous versions of KSQL

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json [hunk 4]
```diff
@@ -112,15 +148,34 @@
...
     },
+    {
+      "name": "stream | initially null | group by (-) | key in value | no aliasing | legacy downstream query",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', value_format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT foo, COUNT(*) FROM INPUT GROUP BY foo;",
+        "CREATE TABLE DOWNSTREAM AS SELECT foo, COUNT(*) FROM OUTPUT GROUP BY foo;"
+      ],
+      "inputs": [
+        {"topic": "OUTPUT", "key": "1", "value": {"FOO":1, "KSQL_COL_1": 1}}
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/key-field.json]: At the moment, nothing. In the new PR that fixes these issues one will have the new corrected functionality and state and one will maintain the legacy functionality and state.

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json [hunk 5]
```diff
@@ -158,15 +232,34 @@
       ],
       "post": {
         "issues": [
-          "key field of output has INCORRECT name: should be 'ALIASED'.",
-          "key field of output has INCORRECT type: should be 'INT'."
+          "key field of output has INCORRECT name: should be 'ALIASED' - (see Issue #1 above)",
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/key-field.json]: Yeah, I have that change locally.  If you don't mind I'll just include it in the next PR.

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json (hunk 1 above)
Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/key-field.json]: QTT only starts the last topology / statement, so you need to generate input for that topology.

File: ksql-engine/src/test/resources/query-validation-tests/key-field.json (hunk 3 above)
Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/key-field.json]: It's needed to create a source table with no key set. I've already documented this locally. If you don't mind, I'll add this in the next PR.

**Pull request merged**